lib/client/weblogin.go

```diff
@@ -434,7 +434,7 @@ func HostCredentials(ctx context.Context, proxyAddr string, insecure bool, req a
 		return nil, trace.Wrap(err)
 	}
 
-	resp, err := clt.PostJSON(ctx, clt.Endpoint("webapi", "host", "credentials"), req)
+	resp, err := clt.PostJSONWithFallback(ctx, clt.Endpoint("webapi", "host", "credentials"), insecure, req)
```

Comment:
Can you tell me who calls `HostCredentials` and why?

Reply:
It's called on the node's initial registration with the cluster, via `TeleportProcess.firstTimeConnect()`. See ./lib/service/connect.go:372